docs/component_spec.md

```diff
@@ -0,0 +1,173 @@
+# Cytoxnet Component Specification
+The package is intended as an exploration and a toolbox for the prediction of
+cytotoxicity of compounds, but aims to leverage datasets of other targets
+for transfer learning.
+***
+***
+## package format
+```
+/cytoxnet
+|-/dataprep
+| |-io.py
+| |-featurize.py
+| |-dataprep.py
+| |-analyze.py
+|-/models
+| |-models.py
+| |-opt.py
+| |-analyze.py
+|-/tests
+```
+***
+
+## `cytoxnet/dataprep/`
+
+### `cytoxnet/dataprep/io.py`
+
+```
+load_file(datafile: str,
+          cols: str or list of str = None) -> pandas DataFrame
+```
+- __Use Case__: (1)
+- __Inputs__: File location string, columns to keep
+- __Outputs__: Dataframe of dataset on file
+
+```
+load_XXX(prepare: bool = False)
+```
+- __Use Case__: (1)
+- __Inputs__: Whether or not to featurize and prepare set using a default
+  pipeline
+- __Outputs__: Dataframe of package dataset
+
+### `cytoxnet/dataprep/featurize.py`
+
+```
+clean_dataframe(dataframe,
+                cols: str or list of str = None,
+                duplicates: str = 'drop',
+                nans: str = 'drop') -> pandas DataFrame
+```
+- __Use Case__: (1, 2)
+- __Inputs__: Dataframe, columns to clean, how to handle nans and duplicates.
+- __Outputs__: Dataframe
+
+```
+featurize(dataframe,
```

Comment:
For our data processing pipeline, I think `featurize()` was broken up into at least two functions, `smiles_to_mol()` and `add_circular_fingerprint()` (although the latter would probably have to just "featurize" or "add_features".  Do we want to do the same thing and split it up, or would it be better to have these functionalities combined into one function like we have here?

Reply:
we can maybe it down write a function that does eg. smiles to mol and call it something like 'structure_string_to_mol' to be general and include InChI. But we would want to call that function within featurize - there is no reason to expose the user to the `Mol` object directly so we might as well have smiles -> feature all in one function